backend/core/src/main/java/org/sonarsource/sonarlint/core/reporting/FindingReportingService.java

```diff
@@ -67,19 +68,21 @@ public class FindingReportingService {
   private final NewCodeService newCodeService;
   private final SeverityModeService severityModeService;
   private final AiCodeFixService aiCodeFixService;
+  private final TelemetryService telemetryService;
   private final PreviouslyRaisedFindingsRepository previouslyRaisedFindingsRepository;
   private final Map<URI, Collection<TrackedIssue>> issuesPerFileUri = new ConcurrentHashMap<>();
   private final Map<URI, Collection<TrackedIssue>> securityHotspotsPerFileUri = new ConcurrentHashMap<>();
   private final Map<String, Alarm> streamingTriggeringAlarmByConfigScopeId = new ConcurrentHashMap<>();
   private final Map<UUID, Set<URI>> filesPerAnalysis = new ConcurrentHashMap<>();
 
   public FindingReportingService(SonarLintRpcClient client, ConfigurationRepository configurationRepository, NewCodeService newCodeService, SeverityModeService severityModeService,
-    AiCodeFixService aiCodeFixService, PreviouslyRaisedFindingsRepository previouslyRaisedFindingsRepository) {
+    AiCodeFixService aiCodeFixService, PreviouslyRaisedFindingsRepository previouslyRaisedFindingsRepository, TelemetryService telemetryService) {
     this.client = client;
     this.configurationRepository = configurationRepository;
     this.newCodeService = newCodeService;
     this.severityModeService = severityModeService;
     this.aiCodeFixService = aiCodeFixService;
+    this.telemetryService = telemetryService;
```

Comment:
Same thing here. You could raise an event when publishing issues, and the count should be made in the telemetry service. The purpose is also to decouple things